File: flutter/packages/measure_flutter/lib/src/exception/exception_severity.dart
import 'package:json_annotation/json_annotation.dart';

/// The severity of an exception.
enum ExceptionSeverity {
  /// An unhandled exception or ANR that crashed the app.
  /// Not produced by the Flutter SDK; reserved for schema parity.
  @JsonValue('fatal')
  fatal,

  /// An exception tracked explicitly by the app using the public API.
  @JsonValue('handled')
  handled,

  /// An unhandled exception that did not crash the app.
  @JsonValue('unhandled')
  unhandled,
}

File: android/measure-android/measure/src/main/java/sh/measure/android/exceptions/ExceptionSeverity.kt
package sh.measure.android.exceptions

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * The severity of an exception.
 */
@Serializable
internal enum class ExceptionSeverity {
    /**
     * An unhandled exception or ANR that crashed the app.
     */
    @SerialName("fatal")
    Fatal,

    /**
     * An exception tracked explicitly by the app using the public API.
     */
    @SerialName("handled")
    Handled,

    /**
     * An unhandled exception that did not crash the app.
     */
    @SerialName("unhandled")
    Unhandled,
}
